File: docs/paydopay-v4/app/Http/Controllers/Api/V1/HealthController.php
<?php

namespace App\Http\Controllers\Api\V1;

use App\Http\Controllers\Controller;
use Illuminate\Http\JsonResponse;
use Illuminate\Support\Facades\DB;

/**
 * GET /api/v1/health
 * Public health endpoint — auth gerekmez.
 * Yanıt: { status, services{ api, database }, time, version }
 */
class HealthController extends Controller
{
    public function index(): JsonResponse
    {
        // DB ping
        $dbStatus = 'ok';
        $dbLatency = null;
        try {
            $t0 = microtime(true);
            DB::select('SELECT 1');
            $dbLatency = (int) round((microtime(true) - $t0) * 1000);
        } catch (\Throwable $e) {
            $dbStatus = 'error';
        }

        $overall = $dbStatus === 'ok' ? 'ok' : 'down';

        // Public response — sistem profili sızdırmamak için minimal payload
        $payload = [
            'status'   => $overall,
            'services' => [
                'api'      => ['status' => 'ok'],
                'database' => ['status' => $dbStatus],
            ],
            'time'     => now()->toIso8601String(),
            'version'  => config('app.version', 'v1'),
        ];

        $http = $overall === 'ok' ? 200 : 503;
        return response()->json($payload, $http);
    }
}

File: src/backend/TPanel.Api/Controllers/V1/HealthController.cs
using System.Diagnostics;
using Dapper;
using Microsoft.AspNetCore.Mvc;
using TPanel.Application.Common.Interfaces;

namespace TPanel.Api.Controllers.V1;

/// <summary>
/// Public sağlık kontrolü — Laravel: GET /api/v1/health (auth gerekmez).
/// </summary>
[ApiController]
[Route("api/v1/health")]
public class HealthController : ControllerBase
{
    private readonly IDbConnectionFactory _connectionFactory;

    public HealthController(IDbConnectionFactory connectionFactory)
    {
        _connectionFactory = connectionFactory;
    }

    [HttpGet]
    public async Task<IActionResult> Index(CancellationToken ct)
    {
        var dbStatus = "ok";
        long latencyMs = 0;
        var sw = Stopwatch.StartNew();
        try
        {
            using var conn = await _connectionFactory.CreateOpenConnectionAsync(ct);
            await conn.ExecuteScalarAsync<int>("SELECT 1");
            latencyMs = sw.ElapsedMilliseconds;
        }
        catch
        {
            dbStatus = "error";
            latencyMs = sw.ElapsedMilliseconds;
        }

        var overall = dbStatus == "ok" ? "ok" : "down";
        var payload = new
        {
            status = overall,
            services = new
            {
                api = new { status = "ok" },
                database = new { status = dbStatus, latency_ms = latencyMs },
            },
            time = DateTime.UtcNow.ToString("yyyy-MM-ddTHH:mm:ssZ"),
            version = "v1",
        };

        return StatusCode(overall == "ok" ? 200 : 503, payload);
    }
}
